src/components/Card.js

```diff
@@ -1,71 +1,85 @@
-
 export default class Card {
-  constructor({ name, link }, cardSelector, handleCardClick) {
+  constructor(
+    { name, link, likes, _id, owner },
+    cardSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    userId
+  ) {
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._cardLikes = likes;
+    this._userId = userId;
+    this._cardId = _id;
+    this._ownerId = owner._id;
   }
 
   _getTemplate() {
     return document
       .querySelector(this._cardSelector)
       .content.firstElementChild.cloneNode(true);
   }
-  _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__button")
-      .addEventListener("click", () => {
-        this._toggleLike();
-      });
 
+  // Event Listeners
+  _setEventListeners() {
     this._cardElement
       .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._deleteCard();
+      .addEventListener("click", (evt) => {
+        evt.preventDefault();
+        this._handleDeleteClick(this._cardId);
       });
-  //   this._cardElement.querySelector(".card__image")
-  //     .addEventListener("click", () => {
-  //       this._imagePopup();
-  //     });
-  // }
+
     this._cardImageEl.addEventListener("click", () => {
-      this._handleCardClick({name: this._name, link: this._link});
-      // popupImage.open(this._cardImageEl);
-      // popupImage.open();
+      this._handleCardClick({ name: this._name, link: this._link });
     });
-    // this._handleCardClick
-  }
+    const cardLikeButton = this._cardElement.querySelector(".card__button");
 
-  // _imagePopup() {
-  //   const imagePopup = document.querySelector("#modal-preview-image");
-  //   this._cardImageEl = imagePopup.querySelector(".modal__image");
-  //   this._cardTitleEl = imagePopup.querySelector(
-  //     ".modal__preview-caption"
-  //   );
-  //   this._cardImageEl.src = this._link;
-  //   this._cardImageEl.alt = this._name;
-  //   this._cardTitleEl.textContent = this._name;
-
-  //   openModal(imagePopup);
-  // }
+    cardLikeButton.addEventListener("click", (evt) => {
+      evt.preventDefault();
+      this._handleLikeClick(this);
+    });
+  }
 
-  _deleteCard() {
+  deleteCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
+  updateLike(result) {
+    this._cardLikes = result.likes;
+    this.revealCardLikes();
+  }
 
-  _toggleLike() {
-    this._cardElement
-      .querySelector(".card__button")
-      .classList.toggle("card__like-button_active");
+  cardLiked() {
+    return this._cardLikes.some((likes) => {
+      return likes._id === this._userId;
+    });
+  }
+
+  revealCardLikes() {
```

Comment:
The logic for rendering likes is brilliant!

Reply:
sounds brilliant coming from you